app/services/statuses_search_service.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class StatusesSearchService < BaseService
+  def call(query, account = nil, options = {})
+    @query   = query&.strip
+    @account = account
+    @options = options
+    @limit   = options[:limit].to_i
+    @offset  = options[:offset].to_i
+
+    status_search_results
+  end
+
+  private
+
+  def status_search_results
+    definition = parsed_query.apply(
+      StatusesIndex.filter(
+        bool: {
+          should: [
+            publicly_searchable,
+            non_publicly_searchable,
+          ],
+          minimum_should_match: 1,
+        }
+      )
+    )
+
+    definition = definition.filter(term: { account_id: @options[:account_id] }) if @options[:account_id].present?
+
+    if @options[:min_id].present? || @options[:max_id].present?
+      range      = {}
+      range[:gt] = @options[:min_id].to_i if @options[:min_id].present?
+      range[:lt] = @options[:max_id].to_i if @options[:max_id].present?
+      definition = definition.filter(range: { id: range })
+    end
+
+    definition.instance_variable_get(:@parameters)[:indices].value[:indices] << PublicStatusesIndex
```

Comment:
I would add a comment here saying this is the only way to perform a multi-index query through Chewy.

Reply:
Updated in https://github.com/mastodon/mastodon/pull/26344/commits/220852a6f3a3fef1f0b737c3096a8d7909f4a9f9